python/lsst/daf/persistence/swiftStorage.py

```diff
@@ -0,0 +1,519 @@
+        a temporary file. If storage is local it may be the original file or
+        a temporary file. The file name can be gotten via the 'name' property
+        of the returned object.
+
+        Raises
+        ------
+        RuntimeError
+            If there is an error downloading the object.
+        """
+        self._log.info("SwiftStorage getting file {}".format(location))
+        start = time.time()
+        try:
+            f = self._getLocalFile(location)
+            self._log.info(
+                "...getting file took {} seconds.".format(time.time() - start))
+            return f
+        except self.swift.ClientException:
+            raise RuntimeError(
+                "Could not download object '{0}/{1}' not found.".format(
+                    self._containerName, location))
+
+    def exists(self, location):
+        """Check if location exists.
+
+        Parameters
+        ----------
+        location : ButlerLocation or string
+            A a string or a ButlerLocation that describes the location of an
+            object in this storage.
+
+        Returns
+        -------
+        bool
+            True if exists, else False.
+        """
+        if isinstance(location, ButlerLocation):
+            location = location.getLocations()[0]
+        return bool(self.instanceSearch(location))
+
+    @staticmethod
+    def getFitsHeaderStrippedPath(path):
+        """Get the path with the optional FITS header selector stripped off.
+
+        Parameters
+        ----------
+        path : string
+            A file path that may end with [n]
+
+        Returns
+        -------
+        (string, string)
+            Tuple, the first item is the path without the fits header, the
+            second item is the part that was stripped, if any.
+        """
+        strippedPath = path
+        pathStripped = ''
+        firstBracket = path.find("[")
+        if firstBracket != -1:
+            strippedPath = path[:firstBracket]
+            pathStripped = path[firstBracket:]
+        return strippedPath, pathStripped
+
+    def instanceSearch(self, path):
+        """Search for the given path in this storage instance.
+
+        If the path contains an HDU indicator (a number in brackets before the
+        dot, e.g. 'foo.fits[1]', this will be stripped when searching and so
+        will match filenames without the HDU indicator, e.g. 'foo.fits'. The
+        path returned WILL contain the indicator though, e.g. ['foo.fits[1]'].
+
+        Parameters
+        ----------
+        path : string
+            A filename (and optionally prefix path) to search for within root.
+
+        Returns
+        -------
+        string or None
+            The location that was found, or None if no location was found.
+        """
+        # Now search for the path in the root or its parents
+        # Strip off any cfitsio bracketed extension if present
+        strippedPath, pathStripped = self.getFitsHeaderStrippedPath(path)
```

Comment:
OK, I'll practice zen, it'll just take longer to review if I need to focus for a minute on that line :smile_cat: